gobblin-data-management/src/main/java/gobblin/data/management/retention/DatasetRetentionRateLimiter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.retention;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gobblin.runtime.Limiter;
+import gobblin.runtime.RateBasedLimiter;
+
+/***
+ * Singleton class for controlling the hdfs call rate during datasets cleaning.
```

Comment:
I think you want to do global throttling for the entire JVM instance. If this is necessary, you can try using Guice https://github.com/google/guice/wiki/Scopes instead of explicitly dealing with singleton.